extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is not IModelBindingData bindingData)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if (bindingData.Source is not Constants.BlobExtensionName)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var content = GetBindingDataContent(bindingData);
+            var connectionName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.Connection, StringComparison.OrdinalIgnoreCase)).Value;
+            var containerName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.ContainerName, StringComparison.OrdinalIgnoreCase)).Value;
+            var blobName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.BlobName, StringComparison.OrdinalIgnoreCase)).Value;
+
+            if (string.IsNullOrEmpty(connectionName) || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
+            var result = Task.Run(() => ToTargetType(context.TargetType, connectionString, containerName, blobName)).Result;
```

Comment:
Can this function be marked `async` and await this call directly?

Some additional points (but moving to async makes them irrelevant): (1) always use `GetAwait().GetResult()` over `.Result` as the later can deadlock. (2) `Task.Run` is not necessary - it just means post this work to the threadpool. You can remove that wrapper and do `ToTargetType(...).GetAwaiter().GetResult();` directly.

Reply:
That's definitely good to know, thanks for sharing!

I couldn't make it async as then we're not following the interface https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Core/Converters/Converter/IInputConverter.cs

and I wasn't sure the impact of me making changes to that interface 🤔

Reply:
Any contract with a `ValueTask<T>` or `Task<T>` return can be made `async` in the implementation. This has no impact on the contract, it is mostly compile-time directives, callers are not aware of `async` or not, they only care about the return type..

Reply:
@liliankasem This existing converter uses async await. https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Core/Converters/JsonPocoConverter.cs

Reply:
Yup its working now, why I was getting interface errors before? I'll never know 😫